vsce/client/src/util/spawnVersionCompareJson.ts

```diff
@@ -0,0 +1,51 @@
+import { spawn } from 'child_process';
+
+export default (
+	pathToScript: string
+): Promise<VersionCompareJson> => {
+	// Track times to find performance bottlenecks.
+	console.info(
+		'Spawned version-compare --json at',
+		new Date().toLocaleTimeString()
+	);
+
+	// https://stackoverflow.com/a/53204227
+	return new Promise((resolve, reject) => {
+		const process = spawn(
+			pathToScript, ['version-compare', '--json']
+		);
+
+		process.on('exit', code => {
+			console.info(
+				'version-compare --json finished...',
+				process,
+				new Date().toLocaleTimeString()
+			);
+
+			const { stdout } = process;
+
+			console.info(
+				'version-compare --json exited ',
+				'with status code',
+				code
+			);
+
+			// "exit code 60 means either the script or
+			// the Docker bits are out of date (or both)"
+			if (code === 60) {
+				const json = JSON.parse(
+					stdout.read(stdout.readableLength)
+				);
+				console.info('json!', json);
+				return resolve(json);
+			}
+
+			if (code)
```

Comment:
This seems like a hole to me but perhaps you've chosen it on purpose.

Wouldn't it make sense to `reject` here and somehow bubble the failure up to the user so we can do something about it?

If we `resolve(null)` the GUI will think nothing is wrong and no updates are available. I think we at least want to `console.error` what happened.

Reply:
What do you think of the approach in [755cafc](https://github.com/reach-sh/reach-lang/pull/655/commits/755cafc787d7d950515f5b67d130f4223756e7bb)?

Reply:
`reject` + `catch` is great but this still only logs the exit code and not the error.

How would we help a user troubleshoot if something went wrong?

Reply:
[49aa168](https://github.com/reach-sh/reach-lang/pull/655/commits/49aa1689cb52efe35ecfd6fb23c2a0930a7d46eb) should address this.

Reply:
[0c847d3](https://github.com/reach-sh/reach-lang/pull/655/commits/0c847d32286b65acc16c1675293315113463d9b9) should address this.

Reply:
[1205582](https://github.com/reach-sh/reach-lang/pull/655/commits/1205582b24a39b73ba63527dfd7a8eb116a66962) should address this.